src/schema/posts.ts

```diff
@@ -296,6 +299,49 @@ export type GQLPostAnalyticsHistory = Omit<
   'post' | 'createdAt'
 >;
 
+const formatCampaignV2toV1 = (campaign: CampaignForV1) => ({
+  post: getFormattedBoostedPost(campaign.post),
+  campaign: {
+    budget: campaign.flags.budget ?? 0,
+    clicks: campaign.flags.clicks ?? 0,
+    users: campaign.flags.users ?? 0,
+    spend: campaign.flags.spend ?? 0,
+    impressions: campaign.flags.impressions ?? 0,
+    campaignId: campaign.id,
+    postId: campaign.referenceId,
+    endedAt: campaign.endedAt,
+    startedAt: campaign.createdAt,
+    status: campaign.state,
+  },
+});
+
+const builderForCampaignV1 = (builder: QueryBuilder<CampaignPost>) =>
+  builder
+    .select('c.id', 'id')
+    .addSelect('c.referenceId', 'referenceId')
+    .addSelect('c.createdAt', 'createdAt')
+    .addSelect('c.endedAt', 'endedAt')
+    .addSelect('c.state', 'state')
+    .addSelect('c.flags', 'flags')
+    .addSelect(
+      `(
+        SELECT json_build_object(
+          'id', p1.id,
+          'shortId', p1."shortId",
+          'slug', p1.slug,
+          'image', p1.image,
+          'title', p1.title,
+          'type', p1.type,
+          'sharedTitle', p2.title,
+          'sharedImage', p2.image
+        )
```

Comment:
To avoid creating more types/interfaces since all these are going to be deleted anyway once we remove support for v1 after getting the right amount of adoption.

Reply:
why do we even do it, why are we not just keeping old queries and once v2 adoption goes through we just kill the old endpoints? I am a bit confused with all the back and forth about v1 and v2

Reply:
Also I am pretty sure you can get this query with repo and query builder, don't see why you need this.